webserver/views/user.py

```diff
@@ -98,6 +99,63 @@ def import_data():
                            lastfm_username=lastfm_username)
 
 
+@user_bp.route("/upload", methods=['POST'])
+@login_required
+def upload():
+    if request.method == 'POST':
+        f = request.files['file']
+        if f:
+            augmented_listens = []
+            msb_listens = []
+            try:
+                jsonlist = json.load(f)
+                if not isinstance(jsonlist, list):
+                    raise ValueError
+            except ValueError:
+                raise BadRequest("Not a valid lastfmbackup file.")
+
+            for listen in jsonlist:
+                dic = _convert_to_native_format(listen)
+                _validate_listen(dic)
+                dic['user_id'] = current_user.musicbrainz_id
+                msb_listens.append(dic)
+                if len(msb_listens) >= MAX_ITEMS_PER_MESSYBRAINZ_LOOKUP:
+                    augmented_listens.extend(_messybrainz_lookup(msb_listens))
+                    msb_listens = []
+
+            if msb_listens:
+                augmented_listens.extend(_messybrainz_lookup(msb_listens))
+
+            _send_listens_to_kafka("import", augmented_listens)
+    return render_template('user/import.html', user=current_user)
```

Comment:
When this page is rendered on a `POST` we should include a message telling the user that their data was imported.